lib/ansible/plugins/connection/aws_ssm.py

```diff
@@ -0,0 +1,469 @@
+# Based on the ssh connection plugin by Michael DeHaan
+#
+# Copyright: (c) 2018, Pat Sharkey <psharkey@cleo.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+author:
+- Pat Sharkey (@psharkey) <psharkey@cleo.com>
+- HanumanthaRao MVL (@hanumantharaomvl) <hanumanth@flux7.com>
+- Gaurav Ashtikar (@gau1991 )<gaurav.ashtikar@flux7.com>
+connection: aws_ssm
+short_description: execute via AWS Systems Manager
+description:
+- This connection plugin allows ansible to execute tasks on an EC2 instance via the aws ssm CLI.
+version_added: "2.8"
+requirements:
+- The remote EC2 instance must be running the AWS Systems Manager Agent (SSM Agent).
+- The control machine must have the aws session manager plugin installed.
+- The remote EC2 instance must have the curl installed.
+options:
+  instance_id:
+    description: The EC2 instance ID.
+    vars:
+    - name: instance_id
```

Comment:
All these vars should start with 'ansible_'

Reply:
https://github.com/ansible/ansible/pull/49652/commits/9b926a0f886a2de90dded4e8113e622debc38868 fixes these and the sample inventory above has been updated. So now an inventory file like - 
```
test:
  hosts:
    linux:
      ansible_aws_ssm_instance_id:  i-063b4d30cb2b72ad8
    windows:
      ansible_aws_ssm_instance_id: i-03b2bbacfa4117f96
      ansible_shell_type: powershell
  children:
    us-east-1:
      hosts:
        linux:
      vars:
        ansible_aws_ssm_region: us-east-1
    us-west-2:
      hosts:
        windows:
      vars:
        ansible_aws_ssm_region: us-west-2
  vars:
    ansible_aws_ssm_bucket_name: your-bucket-name
    ansible_connection: aws_ssm
```

will still allow ad-hoc commands to span host OS types and regions -
```
$ ansible test -i ./hosts.yml -m raw -a hostname                   
windows | CHANGED | rc=0 >>
EC2AMAZ-BSGGS1H


linux | CHANGED | rc=0 >>
ip-10-51-1-108.ec2.internal

$
```